components/rollup_node/README.md

```diff
@@ -0,0 +1,70 @@
+# Rollup Node
+
+The consensus module of Optimistic Ethereum.
+
+## Summary
+
+The Rollup Node is a consensus client that determines the latest state of the rollup. It reads state from L1, and possibly the sequencer, to compute the L2 state. The [block generation spec][block-gen] defines the rules by which L1 state is converted into L2 blocks.
+
+## Types of Rollup Nodes
+There are three primary classes of rollup nodes, corresponding to different configurations of the software stack.
+
+1. **Sequencer**: Accepts user-sent and deposit transactions, orders them into L2 blocks, and submits them to L1 in batches. Provides RPC access to pending L2 blocks in advance of L1 submission.
+2. **L1 Verifier**: Watches L1 feeds, reconstructs L2 block inputs, and inserts them into the EE to determine the canonical L2 state.
+3. **Replicating Verifier**: Performs all the functionality of an L1 verifier, but also tracks pending sequencer state.
+
+### Components Used for Each Node Type
+Of the components outlined below, different node types use different combinations of them.
+
+| Node Type \ Component | Execution Engine | Consensus Layer | Sequencer Replicator | Block Producer | Batch Submitter |
+|-----------------------|------------------|----------------------|----------------------|----------------|-----------------|
+| L1 Verifier           | X                | X                    |                      |                |                 |
+| Replicating Verifier  | X                | X                    | X                    |                |                 |
+| Sequencer             | X                | X                    |                      | X              | X               |
+
+## Components
+
+### [Execution Engine][exec-engine]
+
+The execution engine implements the [execution specification][execution-spec].  Most L1 clients will soon be converted into execution engines. The rollup client will communicate to the engine via a bidirectional JSON-RPC interface ([WIP][execution-engine-rpc]).
+
+One of the main goal of the rollup client is to use the execution engine without modification.
+
+All rollup nodes run an EE.
+
+### [Consensus Layer][consensus-layer]
+
+The consensus layer connects to an L1 node, and tracks the feed data comprising of deposits and sequencer batches. Based on the rules of [block generation][block-gen], it is able to compute sets ("epochs") of L2 blocks as a stateless, pure function of ranges of L1 blocks.
+
+Together with the Execution Engine, it uses the Engine API to progress the L2 state as new data comes in from L1. The consensus layer will reorganize the L2 state if it detects a between the EE and what's on L1, and provides a way for other services to subscribe to reorgs.
+
+All rollup nodes run a CL.
+
+### [Sequencer Replicator][sequencer-replicator]
```

Comment:
discussed with proto today @tynes -- my understanding after that is that 
>difference between an L1 verifier and replicating verifier is running with the p2p networking on or off

Is not accurate. My understanding is that the p2p layer defines whether the state syncing (?) can occur faster than via manual insertion, but you still need some process which looks at the sequencer tip, and tells the EE to go there. This is because the Engine API's syncing model is different from current L1 syncing.

However, I think the logic I sketched out in the replicator process is incorrect.  I'm not going to touch this further though, and will let @protolambda pick this up much better than I'll be able to once this PR is merged.

Reply:
But I did reduce the types of nodes to just Verifier and Sequencer, config on Verifiers will differentiate replicating or not.